### PR TITLE
Generalize task cache to also be aware of task source or version for some specific tasks

### LINKED ISSUE #345: Generalize task cache to also be aware of task source (or version) for some specific tasks
Quoting from #309:
> The limited-scope of this feature is available via #341. If there are no name clashes (e.g. if a single version of tasks is installed), addressing tasks by name should now work also in `fractal workflow add-task`. If there are non-unique names, the cache is never written and only IDs can be used.

See mainly https://github.com/fractal-analytics-platform/fractal/issues/309#issuecomment-1324792116 and following comments.

### PR DESCRIPTION
**Pull request by ychiucco**
closes #345

### CODE REVIEW

File: fractal/cmd/_aux_task_caching.py [hunk 1]
```diff
@@ -20,7 +20,16 @@ async def _fetch_task_list(client: AuthClient) -> _TaskList:
     """
     res = await client.get(f"{settings.BASE_URL}/task/")
     task_list = check_response(res, expected_status_code=200)
-    return task_list
+    return [
+        dict(
```
Comment (tcompa) [inline fractal/cmd/_aux_task_caching.py:24]: What about `dict(include=...)`?
(https://docs.pydantic.dev/latest/usage/exporting_models/#advanced-include-and-exclude)

Comment (tcompa) [reply fractal/cmd/_aux_task_caching.py:24]: My bad, this was for Pydantic objects but we only use dictionaries in this function.

File: fractal/cmd/_aux_task_caching.py [hunk 2]
```diff
@@ -58,82 +67,95 @@ async def refresh_task_cache(client: AuthClient) -> list[dict[str, Any]]:
...
+        if version:
+            raise Exception("---")  # FIXME
+        return _id
+    else:
+        # Set paths
+        cache_dir = Path(f"{settings.FRACTAL_CACHE_PATH}").expanduser()
+        cache_file = cache_dir / TASKS_CACHE_FILENAME
+        # If cache is missing, create it
+        if not cache_file.exists():
+            task_list = await refresh_task_cache(client)
```
Comment (tcompa) [inline fractal/cmd/_aux_task_caching.py:152]: if we enter this branch, then we should set `allow_cache_refresh=False` below

**Comment by ychiucco** [comment]
`Coverage` says that
```markdown
fractal/cmd/_aux_task_caching.py

88.57% of new lines are covered (91.91% of the complete file).
Missing lines: 146, 153, 154, 155
```
but lines 153, 154, 155 are actually covered.

**Pull request merged**